test/integration/smoke/test_multiple_public_ip_ranges.py

```diff
@@ -0,0 +1,887 @@
+        # 3. create a port forwarding on acquired IP from new range
+        # 4. Create a firewall rule to open up the port
+        # 5. Test SSH works to the VM
+
+        self.public_ip_range = PublicIpRange.create(
+                                    self.apiclient,
+                                    self.services["publiciprange"]
+                               )
+
+        self.debug("Dedicating Public IP range to the account");
+        dedicate_public_ip_range_response = PublicIpRange.dedicate(
+                                                self.apiclient,
+                                                self.public_ip_range.vlan.id,
+                                                account=self.account.name,
+                                                domainid=self.account.domainid
+                                            )
+        ip_address = PublicIPAddress.create(
+            self.apiclient,
+            self.account.name,
+            self.zone.id,
+            self.account.domainid,
+            self.services["virtual_machine"]
+        )
+        self.cleanup.append(ip_address)
+        self.cleanup.append(self.public_ip_range)
+        # Check if VM is in Running state before creating NAT and firewall rules
+        vm_response = VirtualMachine.list(
+            self.apiclient,
+            id=self.virtual_machine.id
+        )
+
+        self.assertEqual(
+            isinstance(vm_response, list),
+            True,
+            "Check list VM returns a valid list"
+        )
+
+        self.assertNotEqual(
+            len(vm_response),
+            0,
+            "Check Port Forwarding Rule is created"
+        )
+        self.assertEqual(
+            vm_response[0].state,
+            'Running',
+            "VM state should be Running before creating a NAT rule."
+        )
+
+        # Open up firewall port for SSH
+        FireWallRule.create(
+            self.apiclient,
+            ipaddressid=ip_address.ipaddress.id,
+            protocol=self.services["natrule"]["protocol"],
+            cidrlist=['0.0.0.0/0'],
+            startport=self.services["natrule"]["publicport"],
+            endport=self.services["natrule"]["publicport"]
+        )
+
+        # Create PF rule
+        nat_rule = NATRule.create(
+            self.apiclient,
+            self.virtual_machine,
+            self.services["natrule"],
+            ip_address.ipaddress.id
+        )
+
+        try:
+            logger.debug("SSHing into VM with IP address %s with NAT IP %s" %
+                       (
+                           self.virtual_machine.ipaddress,
+                           ip_address.ipaddress.ipaddress
+                       ))
+            self.virtual_machine.get_ssh_client(ip_address.ipaddress.ipaddress)
+        except Exception as e:
+            self.fail(
+                "SSH Access failed for %s: %s" %
+                (self.virtual_machine.ipaddress, e)
+            )
```

Comment:
Lines 213-224 seem like a good candidate to extract to utility function in the utils module.  Verifying that we can connect via SSH to a host seems like a fairly common test task.